### ISSUE TITLE
Preston et al. 2012. Food web including infectious agents for a California freshwater pond. 

### ISSUE THREAD

**Issue by jhpoelen** [jhpoelen/eol-globi-data#105]
Possible new interaction data:

Daniel L. Preston, Sarah A. Orlofske, John P. McLaughlin, and Pieter T. J. Johnson. 2012. Food web including infectious agents for a California freshwater pond. Ecology 93:1760. http://dx.doi.org/10.1890/11-2194.1

**Comment by millerse** [comment]
There appears to be a nice programable set for this dataset: http://esapubs.org/archive/ecol/E093/153/default.htm

**Comment by jhpoelen** [comment]
@millerse I very much agree! I wish we could figure out a way that would allow you look at the tabular files and assign the columns to biologically meaningful concepts such as taxon name, location, life stage, and interaction type. I imagine these file-column assignments live in a file on GitHub and help GloBI to understand how to read the source data. This way, we could still automate the transformation (original source -[:transform]-> GloBI) without having to do any crazy programming. Perhaps you can ask around and see whether folks like @jhammock and @KatjaSchulz are aware of anyone in the community that use similar methods. Let me know if this is a bit too far out of your comfort zone. . . this is a bit of a dreamy rant ; ) .

**Comment by millerse** [comment]
@jhpoelen 
The other option is to create an access database that can combine the information together to a seamless csv file.  There are files on the site that define what each row means.  Assuming that all of the other datasets have the same headers and information, it should be really easy for me to create a form for globi.

**Comment by jhammock** [comment]
Fun! Since this appears to be a static resource, I would hazard a guess that Sarah's suggestion is more practical. 

Hmm... trying to see if I get what you were suggesting, though, Jorrit: in cases where the source of the data updates, we usually handle renewable harvests with PHP scripts like in https://github.com/eliagbayani/eol_php_code/commit/cd154b90d432fc103e574cdea021e22e148808c6

A lot of what you'll find in https://github.com/eliagbayani/eol_php_code/commits/master has to do with this sort of connector. (The EOL data sources vary *a lot*. Eli has scraped plenty of data for us straight out of the source for html pages with values hard coded in. But many of the sources are databases.) Is that the general approach you were getting at, Jorrit?

**Comment by jhpoelen** [comment]
@jhammock Thanks for sharing your experiences . . . the "renewable harvest" approach is the one that GloBI embraces for (semi)-structured data. Reason is not only that data might be updates / corrected, but (perhaps more importantly) being able to re-do the harvest (or ingestion / import) is key to being able to fix or improve a in the import code / ingestion process. Over the coarse of this project, data issues were quickly resolved because of the level of automation that GloBI provides. 

@millerse I really like your pragmatic approach to map csv files using a access database. How would you (or anybody else) redo this operation in case we find some transformation issue? It would be great if we could automate this. . . . so that we don't have to worry too much about making mistakes ; )

**Comment by cmungall** [comment]
@jhpoelen - related to your dreamy rant - there are a gazillion ETL frameworks for extracting tabular data into some kind of normalized form. Of the subset of these that choose RDF as their target representation I quite like WebKarma. But if i was doing this myself I'd almost always script it, there are usually so many tweaks that need done (in my experience) you need to fall back from the declarative framework to messy imperative coding that you may as well program the thing using some shared programming library of common utilities.

**Comment by jhpoelen** [comment]
@cmungall I have similar experience ETL (extract, transform and load) frameworks - usually just too many exceptions between different data sources . . . but I find it very tempting to believe that there's some magical solution out there. . . how about something along the lines of having a scriptable dataset along the lines of: https://github.com/globalbioticinteractions/template-scriptable-dataset/ . The idea of the newly created template repository is that GloBI would push the defined resources into the custom implementation of importData function and evaluate the result.

**Comment by millerse** [comment]
@jhpoelen 

I am willing to learn some coding so that I can see if I make your idea happen.  I will take a look around.  The only hiccup would be making sure that all the datasets from that site are in the same format with the same headers.

**Comment by jhpoelen** [comment]
Good to hear that your are willing to learn some coding. I do realize that the example I have might be a bit unclear . . . let me think about it a bit more. Ideally, we'd want to have something that allows for easy mapping of columns to standard terms (taxon names, ids, locations, references).

**Comment by jhpoelen** [comment]
@millerse @jhammock @cmungall On inspection, it turned out that this and three other issues #106 #107 #108 were using a file structured similar to #50 . So, I reused the importer and enabled usage through the "hechinger" format: a practical example of why data format standardization is a good idea ; ) . I propose to let the automation discussion simmer until we hit another structured dataset - perhaps we'll get a brilliant insight on how to make it easier to contribute automated data transformations of structured datasets. Will close this an other issues once they make it into http://globalbioticinteractions.org . Thanks for listening!

**Comment by millerse** [comment]
@jhpoelen I did notice that those other issues shared a very similar structure.  For what it is worth, I was able to create an Access database and get all of the nodes and links connected.  The result was that all the datasets were linked into a single file.  There are still some things that need to be worked out for the import to be perfectly automated (such as all of the sheets having a tiny variation in the name that would cause a hiccup), but it is certainly possible.

I will link you to my first attempt with the Acccess table I created: https://github.com/millerse/ESA-Github/blob/master/first%20try.tsv
